mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/index/impl/MongoDBIndexType.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.index.impl;
+
+import org.hibernate.ogm.options.shared.IndexOption;
+import org.hibernate.ogm.util.impl.StringHelper;
+
+/**
+ * Supported MongoDB index types. Defined as a "_type" entry in {@link IndexOption}.
+ *
+ * @author Guillaume Smet
+ */
+public enum MongoDBIndexType {
+
+	NORMAL(null),
+
+	/**
+	 * Full text index.
+	 */
+	TEXT("text"),
+
+	/**
+	 * Geospatial index on a 2D sphere.
+	 */
+	TWODSPHERE("2dsphere"),
+
+	/**
+	 * Geospatial index on a 2D plan.
+	 */
+	TWOD("2d");
+
+	private String name;
```

Comment:
I think this is the worst name you could use :)

It's easy to confuse with the `name()`

Reply:
Maybe `value` or `type`?